src/classes/ModuleErrors.ts

```diff
@@ -1,13 +1,75 @@
+import { colors } from "../../deps.ts";
+import Utils from "./Utils.ts";
 /**
  * a class to display the errors inside the module
  */
-export abstract class ModuleErrors {
-  static checkDiagnostics(diagnostics: any[]) {
-    if (diagnostics) {
-      // TODO expose to the end user diagnostics here
-      console.warn(diagnostics)
+interface ModuleErrorsDiagnostic {
+  start?: {
+    character: number;
+    line: number;
+  };
+  end?: {
+    character: number;
+    line: number;
+  };
+  sourceLine?: string;
+  messageText?: string;
+  messageChain?: {
+    messageText: string;
+    category: number;
+    code: number;
+    next: Pick<ModuleErrorsDiagnostic, 'messageText' | 'category' | 'code'>[];
+  }
+  fileName?: string;
+  category: number;
+  code: number;
+}
+export abstract class ModuleErrors extends Utils {
+  static checkDiagnostics(diagnostics: unknown[]) {
```

Comment:
Typescript compiler provides an inbuilt method to pretty-print colored diagnostics. `ts.formatDiagnosticsWithColorAndContext`. Maybe port it here to match the "official" error reporting style?

Reply:
Where can I get this method natively
I mean is it built-in Deno ?

Reply:
No, it's not. We can try porting it though. Not a priority :+1:

Reply:
Let’s try porting it